pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -409,8 +409,13 @@ func (c *Controller) InstancesByPort(hostname model.Hostname, reqSvcPort int,
 
 // GetProxyServiceInstances returns service instances co-located with a given proxy
 func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.ServiceInstance, error) {
-	var out []*model.ServiceInstance
 	proxyIP := proxy.IPAddress
+	proxyNS := ""
+	if pod, exists := c.pods.getPodByIP(proxyIP); exists {
+		proxyNS = pod.Namespace
+	}
+	endpointsForPodInSameNS := make([]*model.ServiceInstance, 0)
+	endpointsForPodInDifferentNS := make([]*model.ServiceInstance, 0)
```

Comment:
why do we need the second one?

Reply:
Please see my reply in your other comment.